### PR TITLE
Fix error in FZFSinkLocation when filename contains ':'

### PR DESCRIPTION
**Pull request by cheuk-fung**
In function languageClient_FZFSinkLocation, ':' is used to split the
selected location to get tokens for [filename, line, character]. But if
filename contains ':', the length of tokens will be greater than 3. This
fix joins tokens[..(len - 2)] back with ':' to form the correct token
vector [filename, line, character].

### CODE REVIEW

File: src/language_server_protocol.rs [hunk 1]
```diff
@@ -894,6 +894,16 @@ impl LanguageClient {
         Ok(())
     }
 
+    fn edit(&self, goto_cmd: &Option<String>, path: impl AsRef<Path>) -> Fallible<()> {
+        let path = path.as_ref().to_string_lossy();
+        if path.starts_with("jdt://") {
```
Comment (autozimu) [inline src/language_server_protocol.rs:899]: What is the reason for this change? 

I assume it should be addressed in https://github.com/autozimu/LanguageClient-neovim/pull/867

Comment (cheuk-fung) [reply src/language_server_protocol.rs:899]: #867 works only if `locations` contain one element. If there are more elements in `locations`, selection ui will pop up. For example, on a Java interface (eg `List`) that has multiple implementations, call `LanguageClient#textDocument_implementation()` will pop up the selection ui.

For FZF, the `FZFSinkLocation` function will be called after user selects a location. Unfortunately, two issues in `FZFSinkLocation` will prevent displaying jdt content:
- jdt uri contains `:`, which is addressed in the pull request description
- `FZFSinkLocation` calls `self.vim()?.edit()` to load jdt uri, same issue as #867. `self.edit()` function is added to fix this issue. All invocation to `self.vim()?.edit()` is updated to use this function instead so jdt uri can be handled correctly for all use cases.

I hope that answered your question. Thanks.

Comment (autozimu) [reply src/language_server_protocol.rs:899]: 👍
